source/text/Format.cpp

```diff
@@ -44,6 +47,91 @@ namespace {
 		reverse(result.begin(), result.end());
 	}
 
+	void ScanForSubstitutions(const string &source,
+			function<void(const string &, size_t, size_t)> ProcessText,
+			function<const string *(const string &)> SubstitutionFor)
```

Comment:
I see this method called in two places.
The functions passed as `ProcessText` seem basically identical.

From `ExpandInto`:
```c++
		auto ProcessText = [&](const string &from, size_t start, size_t length)
		{
			newValue.append(from, start, length);
		};
```
In `Format::Expand`:
```c++
	auto ProcessText = [&](const string &from, size_t start, size_t length)
	{
		result.append(from, start, length);
	};
```

So, effectively, every time `ScanForSubstitutions` calls `ProcessText`, it is appending to some string.
Why not just have a local string in `ScanForSubstitutions` that gets appended to and then return that at the end?

Reply:
Good catch. I replaced ProcessText with a string return value. Now that the function does more than scan, I renamed it to StringSubstituter